src/algebra/order/nonneg/field.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2021 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import algebra.order.archimedean
+import algebra.order.nonneg.ring
+
+/-!
+# Semifield structure on the type of nonnegative elements
+
+This file defines instances and prove some properties about the nonnegative elements
+`{x : α // 0 ≤ x}` of an arbitrary type `α`.
+
+Currently we only state instances and states some `simp`/`norm_cast` lemmas.
+
+When `α` is `ℝ`, this will give us some properties about `ℝ≥0`.
+
+## Main declarations
+
+* `{x : α // 0 ≤ x}` is a `canonically_linear_ordered_semifield` if `α` is a `linear_ordered_field`.
+
+## Implementation Notes
+
+Instead of `{x : α // 0 ≤ x}` we could also use `set.Ici (0 : α)`, which is definitionally equal.
+However, using the explicit subtype has a big advantage: when writing and element explicitly
+with a proof of nonnegativity as `⟨x, hx⟩`, the `hx` is expected to have type `0 ≤ x`. If we would
+use `Ici 0`, then the type is expected to be `x ∈ Ici 0`. Although these types are definitionally
+equal, this often confuses the elaborator. Similar problems arise when doing cases on an element.
+
+The disadvantage is that we have to duplicate some instances about `set.Ici` to this subtype.
```

Comment:
This doesn't belong in both files, it should only be in `ring.lean`

Reply:
I wasn't quite sure.